wgpu-hal/src/gles/queue.rs

```diff
@@ -0,0 +1,724 @@
+use super::Command as C;
+use arrayvec::ArrayVec;
+use glow::HasContext;
+use std::{mem, ops::Range, slice};
+
+const DEBUG_ID: u32 = 0;
+
+fn extract_marker<'a>(data: &'a [u8], range: &Range<u32>) -> &'a str {
+    std::str::from_utf8(&data[range.start as usize..range.end as usize]).unwrap()
+}
+
+fn is_3d_target(target: super::BindTarget) -> bool {
+    match target {
+        glow::TEXTURE_2D_ARRAY | glow::TEXTURE_3D => true,
+        _ => false,
+    }
+}
+
+impl super::Queue {
+    unsafe fn reset_state(&self) {
+        let gl = &self.shared.context;
+        gl.use_program(None);
+        gl.bind_framebuffer(glow::FRAMEBUFFER, None);
+        gl.polygon_offset(0.0, 0.0);
+        gl.disable(glow::DEPTH_TEST);
```

Comment:
Are resetting viewports and scissors required here?

gl.viewport(0, 0, 0, 0);
gl.depth_range_f32(0.0, 1.0);
gl.scissor(0, 0, 0, 0);

Reply:
Since we set them every time we enable the relevant states, I don't think we need to set the values.
Speaking of which, `polygon_offset` is a leftover here, I'll remove it.